quodlibet/quodlibet/ext/events/synchronizedlyrics.py

```diff
@@ -179,78 +215,184 @@ def _style_lyrics_window(self):
+                    keep_reading = False
+                    word = raw_file[start_ELRC:]
             else:
-                line = raw_file[begin:next_find]
-            begin = next_find
+                word = raw_file[start_ELRC:next_ELRC]
+
+            begin = next_line
+            beginELRC = next_ELRC
 
             # parse lyricsLine
-            if len(line) < 2 or not line[1].isdigit():
+            if len(word) < 2 or not word[1].isdigit():
                 continue
-            close_bracket = line.find("]")
-            t = datetime.strptime(line[1:close_bracket], '%M:%S.%f')
+            if bracketType == 0:
+                close_bracket = word.find("]")
+                line = raw_file[start_ELRC:next_line]
+            if bracketType == 1:
+                close_bracket = word.find(">")
+            t = datetime.strptime(word[1:close_bracket], '%M:%S.%f')
             timestamp = (t.minute * 60000 + t.second * 1000 +
                          t.microsecond / 1000)
-            words = line[close_bracket + 1:]
+            if wentToNextLine:
+                wentToNextLine = False
+                stripELRC = True
+                newLineToStrip = line[close_bracket + 1:]
+                currentStrip = newLineToStrip
+                beginStrip = 0
+                while stripELRC:
+                    startStrippingELRC = newLineToStrip.find("<", beginStrip)
+                    endStrippingELRC = newLineToStrip.find(">",
+                        beginStrip + 1)
+                    strippedLine = currentStrip.replace(
+                    newLineToStrip[startStrippingELRC:endStrippingELRC + 1],
+                        "")
+                    currentStrip = strippedLine
+                    beginStrip = endStrippingELRC
+                    if startStrippingELRC and endStrippingELRC == -1:
+                        stripELRC = False
+                        thisLine = strippedLine
+            words = word[close_bracket + 1:]
             if not words:
+                compressedWords.append(timestamp)
+            else:
+                tmp_word_dict[timestamp] = words
+                for t in compressedWords:
+                    tmp_word_dict[t] = words
+                compressedWords = []
+            if not thisLine:
                 compressed.append(timestamp)
             else:
-                tmp_dict[timestamp] = words
+                tmp_dict[timestamp] = thisLine
                 for t in compressed:
-                    tmp_dict[t] = words
+                    tmp_dict[t] = thisLine
                 compressed = []
 
         keys = list(tmp_dict.keys())
+        wordKey = list(tmp_word_dict.keys())
         keys.sort()
+        wordKey.sort()
         for key in keys:
             self._lines.append((key, tmp_dict[key]))
+        for key in wordKey:
+            self._words.append((key, tmp_word_dict[key]))
         del keys
         del tmp_dict
+        del wordKey
+        del tmp_word_dict
```

Comment:
All of these `del`'s are unnecessary since the variables are local anyway.

https://stackoverflow.com/questions/39255371/when-am-i-supposed-to-use-del-in-python